daemon/pool.go

```diff
@@ -747,21 +795,36 @@ func (c *atomicInt) Value() int {
 // ScalingPolicy specifies whether instances should be started or stopped to
 // cope with load.
 type ScalingPolicy interface {
-	// Scale takes the number of total instances and the load. The load is
-	// the number of request waiting or, there is none, it is a negative
-	// value indicating how many instances are ready.
-	Scale(total, load int) int
+	// Scale takes the total number of active instances, number idle instances and the
+	// number of requests waiting to get a driver instance. Idle may not be zero even if
+	// number of waiting requests is non-zero.
+	// Scale returns the new target number of instances to keep running. This number must
+	// not be less than 1.
+	Scale(total, idle, waiting int) int
+}
+
+// defaultScalingPolicy is the same as DefaultScalingPolicy, but has no window.
```

Comment:
Can you say a bit more in the comment about why we would want this (i.e., no-window)?

Reply:
Sorry, but is this really necessary? Finding usages for this function clearly shows that there are only two of them: one that uses it in the `DefaultScalingPolicy` with a window, and the second one in the tests. There are few tests for the window itself, but there were none for this combination.

Reply:
Is this one here only for the test? If so, can we simply move it into the `_test.go` file so that it doesn't have to pollute the main package? (It's fine for the tests to add things to the package to support the test, if that's all that is really going on here).

Reply:
The test is in the same package, so it should not be possible to update one without the other: The test wouldn't compile. All I'm really saying is that we shouldn't put code that is only used for the test into the part of the package we ship to dependencies, if possible.

Reply:
We chatted about this in Slack: The separation is to make it easier to test, but is not only for the test (I was confused). We can leave this alone for now and think about how to test it more completely as we refine it.